artifactory/commands/generic/upload.go

```diff
@@ -209,3 +252,18 @@ func getUploadParams(f *spec.File, configuration *utils.UploadConfiguration) (up
 
 	return
 }
+
+func (uc *UploadCommand) handleSyncDelete(syncDeleteProp string) error {
+	var deleteSpec = createDeleteSpecForSync(uc.SyncDeletePath(), syncDeleteProp)
+	deleteCommand := NewDeleteCommand()
```

Comment:
We are trying to avoid commands invoking other commands. This could cause statistics counting issues as well as other problems.
The code flow should be - commands invoke jfrog-client-go APIs.